src/NServiceBus.RavenDB/Outbox/OutboxRecordsCleaner.cs

```diff
@@ -26,38 +30,28 @@ public void RemoveEntriesOlderThan(DateTime dateTime)
 
             try
             {
-                var deletionCommands = new List<ICommandData>();
+                var query = new IndexQuery
+                {
+                    Query = $"Dispatched:true AND DispatchedAt:[* TO {dateTime:o}]"
+                };
 
-                using (var session = DocumentStore.OpenSession())
+                var bulkOpts = new BulkOperationOptions
                 {
-                    var query = session.Query<OutboxRecord, OutboxRecordsIndex>()
-                        .Where(o => o.Dispatched)
-                        .OrderBy(o => o.DispatchedAt);
-
-                    QueryHeaderInformation qhi;
-                    using (var enumerator = session.Advanced.Stream(query, out qhi))
-                    {
-                        while (enumerator.MoveNext())
-                        {
-                            if (enumerator.Current.Document.DispatchedAt >= dateTime)
-                            {
-                                break; // break streaming if we went past the threshold
-                            }
-
-                            deletionCommands.Add(new DeleteCommandData
-                            {
-                                Key = enumerator.Current.Key
-                            });
-                        }
-                    }
-                }
+                    AllowStale = true
```

Comment:
AllowStale "Indicates whether operations are allowed on stale indexes". In the case of Outbox cleanup, this information is already probably 7 days old. I sure hope it has made it into an index by then.